### DISCUSSION TITLE
Qt4::qtmain target not found.

### DISCUSSION THREAD

**Discussion by freeNastran** [SAP-archive/fedem-gui#28] [General]
Hi ,

Tring to build the GUI and got this error. It found Qt4 and qtmain.lib is in the lib folder.

`-- INFORMATION : Processing unit FEDEM_GUI_vpmDB
-- Found Qt4: D:/FEDEM/fedem-gui-main/3rdLib/bin/qmake.exe (found version "4.8.7")
-- Building library vpmDB
-- Configuring with Qt4
-- Building executable Fedem
-- Configuring done (25.5s)
CMake Error at cmake/Modules/FindQt4.cmake:1101 (set_property):
  The link interface of target "Qt4::QtCore" contains:

    Qt4::qtmain

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  fedem-mdb/vpmDB/CMakeLists.txt:31 (find_package)`

Cheers

**Comment by kmoks** [comment]
Hi, tanks for your contribution, and sorry for delayed reply.

I'm not sure what goes wrong here. But I assume you have verified that the qtmain.lib file exists in the folder `D:/FEDEM/fedem-gui-main/3rdLib/bin/`? Did you build the Qt 4.8.7 package yourself, or did you download the binaries from somewhere. Then you have to make sure they are compatible with your system.

Another thing to check is the file `cmake/Modules/FindQt4.cmake`. This is a modified version of the file that comes with cmake version 3.17 which we have used, to facilitate that the environment variable QT_ROOT can be set to point to where you have the Qt4 installation. If you have a newer cmake version, e.g., the one that comes with VS 2022, there is a chance this file does not work. In that case, you could try to grab the `FindQt4.cmake` file from your installation, and modify it accordingly, and replace the current file with it. You may even try to run without this file by just deleting it first, such that cmake then uses the one embedded with your installation.

Just in case, the difference between the modified file and the original, is as follows:
```
$ diff -b cmake/Modules/FindQt4.cmake /c/Program\ Files/CMake/share/cmake-3.17/Modules/FindQt4.cmake
 4,6d3
 < # NOTE: This is a modified version of the same file which comes with cmake 3.17
 < # and facilitates that Qt can be located through environment variable QT_ROOT.
 <
 338,343c335,338
 < # KMO change 07.12.22: ${CMAKE_CURRENT_LIST_DIR} --> ${CMAKE_ROOT}/Modules
 < include(${CMAKE_ROOT}/Modules/CheckCXXSymbolExists.cmake)
 < include(${CMAKE_ROOT}/Modules/MacroAddFileDependencies.cmake)
 < include(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
 < include(${CMAKE_ROOT}/Modules/CMakePushCheckState.cmake)
 < # end KMO change
 ---
 > include(${CMAKE_CURRENT_LIST_DIR}/CheckCXXSymbolExists.cmake)
 > include(${CMAKE_CURRENT_LIST_DIR}/MacroAddFileDependencies.cmake)
 > include(${CMAKE_CURRENT_LIST_DIR}/FindPackageHandleStandardArgs.cmake)
 > include(${CMAKE_CURRENT_LIST_DIR}/CMakePushCheckState.cmake)
 457,465d451
 <   # KMO change 07.12.22: Use env QT_ROOT for some variables
 <   set(return_code 0)
 <   if(${VAR} STREQUAL "QT_INSTALL_HEADERS")
 <     set(output "$ENV{QT_ROOT}\\include")
 <   elseif(${VAR} STREQUAL "QT_INSTALL_LIBS")
 <     set(output "$ENV{QT_ROOT}\\lib")
 <   elseif (${VAR} STREQUAL "QT_INSTALL_BINS")
 <     set(output "$ENV{QT_ROOT}\\bin")
 <   else()
 470,471d455
 <   endif()
 <   # end KMO change
 1186,1188c1170
 <   # KMO change 07.12.22: Use CMAKE_ROOT directly instead of CMAKE_CURRENT_LIST_FILE
 <   set(_qt4_current_dir "${CMAKE_ROOT}/Modules")
 <   # end KMO change
 ---
 >   get_filename_component( _qt4_current_dir  "${CMAKE_CURRENT_LIST_FILE}" PATH)
```
Hope this helps you along. If you find a solution, please post an update here, or create an Issues if it requires something to be fixed in the code base. Thanks!
